grails-app/controllers/org/pih/warehouse/order/OrderController.groovy

```diff
@@ -83,9 +84,16 @@ class OrderController {
         }
 
         try {
+            def shipmentIds = command.order?.orderItems?.shipmentItems*.shipment*.id.flatten().unique()
```

Comment:
First we need to make sure we only update PENDING shipments. 

In order to make this more DRY, let's add methods to Order and Shipment to getShipments() and getOrders(), respectively. There should be one method that gets all and another that allows you to get by status. 

Something like this. And make sure to add "shipments" to the transients list.
```
List getShipments() { 
    orderItems?.shipmentItems*.shipment*.flatten()?.unique() 
}
List getShipments(ShipmentStatusCode statusCode) { 
    orderItems?.shipmentItems*.shipment.flatten()?.unique().findAll { it.currentStatus == statusCode }
}
```
then this line would be something like this
```
def shipmentIds = command.order?.getShipments(ShipmentStatusCode.PENDING).collect { it.id }
```

Reply:
@jmiranda  Ugh,  i have not noticed that there is already listShipments() in Order. Ill fix that today.

Reply:
@awalkowiak Let me know if this is ready for review again.

Reply:
@jmiranda Ready